Xam.Plugins.OnDeviceCustomVision/ImageClassifier.ios.cs

```diff
@@ -27,7 +27,13 @@ public class ImageClassifierImplementation : ImageClassifierBase
 
         private VNCoreMLModel LoadModel(string modelName)
         {
-            var modelPath = NSBundle.MainBundle.GetUrlForResource(modelName, "mlmodelc") ?? CompileModel(modelName);
+            var isPath = File.Exists(modelName);
+            var isCompiled = Path.GetExtension(modelName).Equals("mlmodelc");
+            NSUrl modelPath = isCompiled
+                ? (isPath
+                    ? new NSUrl(modelName, false)
+                    : NSBundle.MainBundle.GetUrlForResource(modelName, "mlmodelc"))
+                : CompileModel(modelName, isPath);
```

Comment:
Can you clean up this code - I find nested ternary operators hard to grok, so would favor a solution with more readability.

Reply:
Done.
I changed it to nested if-else.